### PR TITLE
reimplement the wait_(screen_still|changed_screen) functions

### PR DESCRIPTION
**Pull request by coolo**
not to check the image ourselves, but to rely on the backend thread to check

### CODE REVIEW

**Comment by okurz** [comment]
+1

I have good experiences stating the reason or benefit of the change in the subject line of the commit and pull requests, i.e. turn around commit subject and description.

File: testapi.pm [hunk 1]
```diff
@@ -279,9 +279,50 @@ sub upload_asset {
...
+    my ($callback) = @_;
+
+    bmwqemu::log_call('wait_screen_change');
+
+    # get the initial screen
+    $bmwqemu::backend->set_reference_screenshot;
+    $callback->() if $callback;
+
+    my $starttime        = time;
+    my $timeout          = 10;
```
Comment (aaannz) [inline testapi.pm]: hard coded timeout? I'm for to have 'timeout => [0-9]+' as option.

Comment (aaannz) [inline testapi.pm]: Or as another positional argument as in `wait_still_screen` lower.

**Comment by aaannz** [comment]
Travis complaining on using prototypes in testapi. When did we remove prototypes from testapi? I thought this was the only place we actually wanted them.

File: testapi.pm (hunk 1 above)
Comment (coolo) [inline testapi.pm]: I have no idea how to express this in the parameter line ;)

**Comment by coolo** [comment]
actually last time I checked you require the prototype for perl to accept code as parameter

**Comment by aaannz** [comment]
Indeed, but I have been thinking, albeit not expressed enough apparently, differently about testapi and I'm in favour of having testapi simulate kind-of-DSL. I surely can live with that, I'm just surprised.

File: testapi.pm (hunk 1 above)
Comment (aaannz) [inline testapi.pm]: `wait_screen_change(sub {...}, $timeout)` wouldn't work?
and
`my ($cb, $timout) = @_;` on other side?

**Comment by coolo** [comment]
it still works kind-of-DSL - all perl requires is that it knows it's a function.

**Comment by coolo** [comment]
what you're lacking without prototypes is the "too few arguments" error

**Comment by coolo** [comment]
added a timeout paramter and shut up perlcritic (hopefully)

**Comment by aaannz** [comment]
Aha, all those years and didn't know that! I needed to try it right now to actually believe you and you were right!

**Pull request merged**